### DISCUSSION TITLE
Google Cloud Spanner Support

### DISCUSSION THREAD

**Discussion by campfireman** [evidence-dev/evidence#2377] [Data Source Request]
## What do you use this database for?

We have a smaller set of data in production, we don't keep a separate data warehouse. Therefore, we perform our analyses directly on our Google Cloud Spanner instance.

## Are there any popular and well-maintained clients for this database?

All popular languages have support, npm ist supported as well: https://www.npmjs.com/package/@google-cloud/spanner

## Are you interested in contributing this database connector?

I haven't had the time to dig into your code. But I see you already support big query, so I assume there is plumbing I could recycle. I'd by happy to contribute! :)

**Comment by archiewood** [comment]
As another question, in the absence of a current Spanner connector, would you be able to use the BQ connector like this?

https://cloud.google.com/bigquery/docs/federated-queries-intro

https://cloud.google.com/bigquery/docs/cloud-spanner-federated-queries

**Comment by archiewood** [comment]
Hello - looping back on connector requests

We have recently re-architected Evidence (since this issue was opened), and Evidence is now "pluggable", meaning you can write your own connectors, publish them, and use them in Evidence.

If you are interesting in writing a connector, here are some resources:

1. Docs on creating a plugin: https://docs.evidence.dev/plugins/creating-a-plugin/datasources
2. Template for a connector plugin: https://github.com/evidence-dev/datasource-template
3. An example that I spun up for pulling stats from the npm API: https://github.com/archiewood/npm-stats

If you do write a plugin:

- the suggested package name is `evidence-connector-[plugin]`
- the docs are new, so let us know if there is extra information you need
- share it on the Evidence [community](https://github.com/evidence-dev/evidence/issues/slack.evidence.dev), and we'll feature it!
